src/Agent.Sdk/Util/IOUtil.cs

```diff
@@ -459,5 +459,26 @@ private static void RemoveReadOnly(FileSystemInfo item)
                 item.Attributes = item.Attributes & ~FileAttributes.ReadOnly;
             }
         }
+
+        public static string GetDirectoryName(string path, PlatformUtil.OS platform)
+        {
+            string retval = Path.GetDirectoryName(path.TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar));
+            if (PlatformUtil.RunningOnWindows)
+            {
+                if (platform != PlatformUtil.OS.Windows)
+                {
+                    retval = retval.Replace("\\","/");
```

Comment:
That is an interesting question on what to do here. The issue was this:

```
Path.GetDirectoryName("/foo/bar/a");
```
On Windows would return:
```
\foo\bar
```
instead of the expected
```
/foo/bar
```

I guess we could just implement the directory name logic ourselves...

Reply:
The more I think about this, I think I'm fine leaving it. I think its reasonable to not allow restricted characters for Windows or Linux for this use case. This won't actively break anyone, if this is causing issues we can fix it. At the very least, I don't think this needs to block this PR, we can follow up if needed.

Reply:
I have a change in a follow up PR that will be pushed imminently so I will go ahead and merge this one.